upload/catalog/controller/extension/module/zasilkovna.php

```diff
@@ -129,9 +129,22 @@ public function sessionCheckOnShippingChange(&$route, &$args)
 	{
 		$oldAddressId = $this->session->data["shipping_address"]["address_id"];
 		$newAddressId = $this->request->post["address_id"];
-		if ($oldAddressId != $newAddressId) {
+
+		if ($oldAddressId != $newAddressId || $this->request->post['shipping_address'] === 'new') {
```

Comment:
$this->request->post['shipping_address']   používáš 3 místech.  Nebylo by lepší si hodnotu "vytáhnout" do proměnné ?